src/test/scala/firrtlTests/SerializerSpec.scala

```diff
@@ -121,4 +121,19 @@ class SerializerSpec extends AnyFlatSpec with Matchers {
     val serialized = Serializer.serialize(when, 1)
     serialized should be("  when cond :\n    skip\n")
   }
+
+  it should "serialize read-under-write behavior for smems correctly" in {
+    def parseSerializeParse(src: String): Circuit = Parser.parse(Parser.parse(src).serialize)
+    val undefined = parseSerializeParse(SMemTestCircuit.src(""))
+    assert(SMemTestCircuit.findRuw(undefined) == ReadUnderWrite.Undefined)
```

Comment:
Nit: using non-assert matchers may be more readable:

```scala
SMemTestCircuit.findRuw(undefined) should be (ReadUnderWrite.Undefined)
```

Reply:
I disagree. I like to keep things simple and scalatest will still highlight the failure correctly when using plain asserts. The `should` or `must` matchers have some weird corner cases (I forgot what they were, but there was some trouble in the past that would have been avoided by sticking to the basics).

Reply:
Let me know if you insist on using the `should` and I will change it in order to get this change in.

Reply:
There's no problem with using them here and continuing to use them aligns with all the other tests in this file.

For corner cases, you may be thinking of string matching on `include` vs `contains`? The former looks for a substring match and the latter looks for an exact match.

Reply:
Don't care. It's good to land as-is. 👍